components/browser/icons/src/main/java/mozilla/components/browser/icons/processor/ColorProcessor.kt

```diff
@@ -0,0 +1,87 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.icons.processor
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Color.rgb
+import androidx.annotation.ColorInt
+import androidx.palette.graphics.Palette
+import mozilla.components.browser.icons.Icon
+import mozilla.components.browser.icons.IconRequest
+import kotlin.math.max
+
+/**
+ * [IconProcessor] implementation to extract the dominant color from the icon.
+ */
+class ColorProcessor : IconProcessor {
```

Comment:
It turned out that Fenix didn't need that and therefore we never implemented that. I wonder if we should split this PR into two and maybe hold back on landing this part?

Reply:
I can split this out, I'll need it for adaptive launcher icons.